components/bpmn/org.wso2.carbon.bpmn.uuf.ui.stub/src/main/java/org.wso2.carbon.bpmn.uuf.ui/service/BPMNExplorerServiceImpl.java

```diff
@@ -0,0 +1,522 @@
+                        deploymentObj.put("process-name", arrayJsonObj.get("name"));
+                        deploymentObj.put("process-def-id", arrayJsonObj.get("id"));
+                        deploymentObj.put("version", arrayJsonObj.get("version"));
+                        deploymentObj.put("image-data",
+                                          getBPMNProcessDiagram(arrayJsonObj.get("id").toString(), cookie));
+                        JSONObject oneProcessObj = new JSONObject();
+                        oneProcessObj.put("oneProcess", deploymentObj);
+                        returnDataArray.put(oneProcessObj);
+                    }
+                }
+
+                JSONArray pagiArray = new JSONArray();
+                for (int i = 1; i <= paginatePageCount; i++) {
+                    JSONObject pageObj = new JSONObject();
+                    pageObj.put("pageNu", i);
+                    pagiArray.put(pageObj);
+
+                }
+                JSONObject pagiObj = new JSONObject();
+                pagiObj.put("paginationCount", pagiArray);
+                returnDataArray.put(pagiObj);
+            }
+
+        } catch (IOException e) {
+            log.error("Exception : ", e);
+        } catch (JSONException e) {
+            log.error("Exception : ", e);
+        } catch (URISyntaxException e) {
+            log.error("Exception : ", e);
+        } finally {
+            if (rd != null) {
+                rd.close();
+            }
+
+            if (response != null) {
+                response.close();
+            }
+
+            if (httpGet != null) {
+                httpGet.releaseConnection();
+            }
+        }
+        return returnDataArray;
+    }
+
+    /**
+     * Return session cookie from Authentication Admin Service
+     */
+    private String HTTPAuthenticate() {
+        HttpPost httpPost;
+        CloseableHttpResponse response;
+        BufferedReader rd;
+        String line;
+        StringEntity stringEntity;
+        String cookie = null;
+
+        try {
+            httpPost = new HttpPost(baseUrl + "/services/AuthenticationAdmin.AuthenticationAdminHttpsSoap11Endpoint/");
+            httpPost.addHeader("SOAPAction", "urn:login");
+            httpPost.addHeader("Content-Type", "text/xml");
+
+            String payload =
+                    "<soapenv:Envelope xmlns:soapenv=\"http://schemas.xmlsoap.org/soap/envelope/\" " +
+                            "xmlns:aut=\"http://authentication.services.core.carbon.wso2" +
+                            ".org\"><soapenv:Header/><soapenv:Body><aut:login><aut:username>admin</aut:username" +
+                            "><aut:password>admin</aut:password></aut:login></soapenv:Body></soapenv:Envelope>";
+
+            stringEntity = new StringEntity(payload, "UTF-8");
+            stringEntity.setChunked(true);
+
+            httpPost.setEntity(stringEntity);
+
+            response = client.execute(httpPost);
+            rd = new BufferedReader(
```

Comment:
can use try with resource statement to auto close streams
https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html